lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -8697,22 +8697,29 @@ namespace Js
                     // compact the types array by moving non-null types
                     // at the beginning.
                     this->types[nonNullIndex++] = type;
-                    isAnyTypeLive = true;
 #if DBG
                     isGuardValuePresent = this->guard->GetValue() == reinterpret_cast<intptr_t>(type) ? true : isGuardValuePresent;
 #endif
+                    this->types[i] = nullptr;
```

Comment:
What if nonNullIndex (before++) == i? You will null out something that we just assigned.